lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
+        return cb(err);
+      }
```

Comment:
@kjdelisle 
> What are the potential pitfalls of using updateAttributes as the default action here

I could not think of how to deal with `updateAttributes` here lol, by `updateOrCreate`, you give cloudant a data without knowing is it exist or not, neither what does the data look in it.
While by 'updateAttributes', you know exactly which properties you want to update. Calling updateAttributes we have to fetch original data then magically "merge" it with our data, then update certain properties...